src/ex_docmd.c

```diff
@@ -5766,6 +5766,93 @@ get_tabpage_arg(exarg_T *eap)
   return tab_number;
 }
 
+/*
+ * Handle the argument for a tabpage related ex command.
+ * Returns a tabpage request.
+ * When an error is encountered then eap->errmsg is set.
+ */
+static tabPageRequest_T
+oni_create_tabpage_request(tabPageKind_T kind, exarg_T *eap)
+{
```

Comment:
This duplicates quite a bit of `get_tabpage_arg`, but removes the checks for valid tab index and everything else I don't understand. We'll also still get nice error messages when the format of the argument is invalid.

Reply:
Looks good to me - the checks for valid tab index can come on the Onivim 2 side (since it's not changing any state in the vim world, here). The logic is easy to follow.

> We'll also still get nice error messages when the format of the argument is invalid.

Perfect!